driver-core/src/main/java/com/datastax/driver/core/querybuilder/BuiltStatement.java

```diff
@@ -231,7 +231,7 @@ void maybeAddRoutingKey(String name, Object value) {
             return;
 
         for (int i = 0; i < partitionKey.size(); i++) {
-            if (name.equals(partitionKey.get(i).getName())) {
+            if (name.equalsIgnoreCase(partitionKey.get(i).getName())) {
```

Comment:
`Metadata.handleId` has package protected access - should i just make it public?